optuna/notebooks/azure-optuna/run_optuna.ipynb

```diff
@@ -0,0 +1,249 @@
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core import Environment\n",
+    "\n",
+    "# create the environment\n",
+    "rapids_env = Environment('rapids_env')\n",
+    "\n",
+    "# create the environment inside a Docker container\n",
+    "rapids_env.docker.enabled = True\n",
+    "\n",
+    "# specify docker steps as a string. Alternatively, load the string from a file\n",
+    "dockerfile = \"\"\"\n",
+    "FROM rapidsai/rapidsai:0.16-cuda10.2-runtime-ubuntu18.04-py3.7\n",
+    "RUN apt-get update && \\\n",
+    "apt-get install -y fuse && \\\n",
+    "apt-get install libssl1.0.0 libssl-dev && \\\n",
+    "source activate rapids && \\\n",
+    "pip install azureml-sdk==1.13.0 && \\\n",
+    "pip install azureml-widgets && \\\n",
+    "pip install optuna && \\\n",
+    "pip install dask_optuna && \\\n",
+    "pip install fusepy\n",
+    "\"\"\"\n",
+    "\n",
+    "# set base image to None since the image is defined by dockerfile\n",
+    "rapids_env.docker.enabled = True\n",
+    "rapids_env.docker.base_image = None\n",
+    "rapids_env.docker.base_dockerfile = dockerfile\n",
+    "\n",
+    "# use rapids environment in the container\n",
+    "rapids_env.python.user_managed_dependencies = True"
+   ]
+  },
+  {
+   "source": [
+    "## Acquiring Data\n",
+    "\n",
+    "1. Download data from Kaggle - [BNP Paribas Carid Calims Management](https://www.kaggle.com/c/bnp-paribas-cardif-claims-management/data)\n",
+    "2. Place only the unzipped `train.csv` file in `data_dir`(by default this is set to `'data/'`)\n",
+    "3. The following cell will upload it to your default workspace datastore under `path_on_datastore`."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.dataset import Dataset\n",
+    "\n",
+    "data_dir = \"data/\"\n",
+    "path_on_datastore = 'bnp_upload'\n",
+    "\n",
+    "datastore.upload(src_dir=data_dir, target_path=path_on_datastore, overwrite=False, show_progress=True)\n",
+    "\n",
+    "ds_data = datastore.path(path_on_datastore)\n",
+    "dataset = Dataset.File.from_files(ds_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "script_params = ['--data_dir', dataset.as_named_input('bnp_input').as_mount(),\n",
+    "]\n",
+    "from azureml.core import ScriptRunConfig\n",
+    "\n",
+    "project_folder = \"./\"\n",
```

Comment:
I'd suggest adding the training script in a folder (for example project_folder), and passing that as the source_directory instead of current one. AzureML will upload a snapshot of source directory, so will throw an error if there are large files